pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -614,6 +615,58 @@ func applyLoadBalancer(cluster *v2.Cluster, lb *networking.LoadBalancerSettings)
 			},
 		}
 	}
+
+	// Locality weighted load balancing
+	cluster.CommonLbConfig = &v2.Cluster_CommonLbConfig{
+		LocalityConfigSpecifier: &v2.Cluster_CommonLbConfig_LocalityWeightedLbConfig_{
+			LocalityWeightedLbConfig: &v2.Cluster_CommonLbConfig_LocalityWeightedLbConfig{},
+		},
+	}
+
+	applyLBLocalityWeight(proxy, cluster.LoadAssignment, lb)
+}
+
+func applyLBLocalityWeight(proxy *model.Proxy, loadAssignment *v2.ClusterLoadAssignment, lb *networking.LoadBalancerSettings) {
+	if proxy == nil || proxy.Locality == "" {
+		return
+	}
+
+	for _, localityWeightSetting := range lb.LocalityWeightSettings {
+		if localityWeightSetting != nil &&
+			util.LocalityMatch(proxy.Locality, localityWeightSetting.From) {
+			misMatched := map[int]struct{}{}
+			for i := range loadAssignment.Endpoints {
```

Comment:
this wont work here as cluster.go is for CDS that does not have endpoint data for EDS clusters. There is endpoint data for static and strictdns clusters though